### ISSUE TITLE
Analyzer to simplify LINQ expressions by using an overload that accepts a Func<TSource,TResult>

### ISSUE THREAD

**Issue by Bartleby2718** [dotnet/roslyn#75845]
**Brief description:**

Some LINQ methods have an overload that accepts as an extra parameter `Func<TSource,TResult>` to transform the result beforehand. We can simplify some LINQ expressions by taking advantage of such overloads.

**Languages applicable:**

Both C# and Visual Basic

**Code example that the analyzer should report:**

```cs
int[] numbers = [1, 2, 3, 4, 5, 6, 7];
// before
var sumOfSquares = numbers.Select(n => n * n).Sum();
// after
var sumOfSquares = numbers.Sum(n => n * n);
```

**Additional information:**

Relevant LINQ methods I can think of:
- `Sum`
- `Min`
- `Max`

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
